batcher/aligned-batcher/src/lib.rs

```diff
@@ -189,12 +200,28 @@ impl Batcher {
         // FIXME: We are not doing anything for the moment with the address from the
         // sender, this logic should be added for the payment system.
         info!("Verifying message signature...");
-        if let Ok(_addr) = client_msg.verify_signature() {
+        let submitter_addr = if let Ok(addr) = client_msg.verify_signature() {
             info!("Message signature verified");
-            // do something with addr
+            let user_balance = self
+                .payment_service
+                .user_balances(addr)
+                .call()
+                .await
+                .unwrap_or_default();
+
+            if user_balance == U256::from(0) {
+                error!("Insufficient funds for address {:?}", addr);
+                return Err(tokio_tungstenite::tungstenite::Error::Protocol(
+                    ProtocolError::HandshakeIncomplete,
+                ));
+            }
```

Comment:
Should probably check against a fixed value > 0

I think in contract it is 0.0004 eth or something like that

Reply:
+1

Reply:
this was added in issue #521 , to solve all of these cases together